python/tests/test_mesh_metric.py

```diff
@@ -0,0 +1,33 @@
+import sys
+import os
+sys.path.append(os.path.abspath('..'))
+from dolfin import *
+from adaptivity import *
+from numpy import array
+
+check = []
+test_metric = array([[1.0, -0.5], [-0.5, 1.0]])
+
+for n in range(2, 20):
+    mesh = UnitSquareMesh(mpi_comm_world(), n,n)
+
+    M = mesh_metric(mesh)
+
+    A = M.vector().array().reshape((mesh.num_cells(),2,2))
+
+    # the metric should be constant on each element
+    if abs(A - A[0,:,:]).max()/A.max() > 1.0e-14:
+        raise RuntimeError('Test failed: the metric is not constant on each element')
+
+    a = A[0,:,:]*mesh.hmax()**2/2.
+
```

Comment:
why do you need to divide by 2 here ?
thanks!

Reply:
No real need. After rescaling by h^2/2 the metric is then always [1, -0.5; -0.5, 1].

Reply:
is h the size of the diagonal of the triangles, or the size of the other edges ? It makes sense in the former case, less in the latter...

Reply:
For the mesh size, mesh.hmax() computes the maximum cell diameter, the diameter is computed as two times the circumradius (http://mathworld.wolfram.com). As the elements have a 90 degrees angle hmax gives the diagonal of the triangles.